dnsdist-0.0.147gd5b4f7f fedora 22 missing lua-devel not detected

if the lua-devel package is not installed it's not properly detected and reported as fatal when building with fedora 22.

eg:

make[2]: Entering directory '/home/jared/dnsdist-0.0.147gd5b4f7f'
  CXX      dnsdist.o
In file included from dnsdist.hh:2:0,
                 from dnsdist.cc:23:
ext/luawrapper/include/LuaContext.hpp:55:19: fatal error: lua.hpp: No such file or directory

installing lua-devel corrects this build issue

Hello.

How I see I have same problem with debian 8.
configure failed with
checking for LUA... no
checking for LUA... no
checking for LUA... no
checking for LUA... no
checking for LUA... no
checking for LUA... no
checking for chosen LUA... not found
checking whether g++ supports C++11 features by default... no
checking whether g++ supports C++11 features with -std=gnu++11... no
checking whether g++ supports C++11 features with -std=gnu++0x... no
configure: error: *** A compiler with support for C++11 language features is required.

without lua headers and I try to fix issue with g++.

But after install liblua5.2-dev it confugure good - 
checking for LUA... no
checking for LUA... yes
checking for chosen LUA... lua5.2
checking whether g++ supports C++11 features by default... no
checking whether g++ supports C++11 features with -std=gnu++11... yes
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating Makefile
config.status: creating pdns/ext/yahttp/Makefile
config.status: creating pdns/ext/yahttp/yahttp/Makefile
config.status: creating config.h
config.status: executing depfiles commands
config.status: executing libtool commands

--- a/pdns/dnsdistdist/configure.ac
+++ b/pdns/dnsdistdist/configure.ac
@@ -13,7 +13,7 @@ BOOST_REQUIRE([1.35])
 BOOST_FOREACH
 AC_SUBST([YAHTTP_CFLAGS], ['-I$(top_srcdir)/ext/yahttp'])
 AC_SUBST([YAHTTP_LIBS], ['-L$(top_builddir)/ext/yahttp/yahttp -lyahttp'])
-PDNS_WITH_LUA
+DNSDIST_LUA
 AX_CXX_COMPILE_STDCXX_11(ext,mandatory)
 
 AC_CONFIG_FILES([Makefile

--- a/pdns/dnsdistdist/m4/dnsdist_lua.m4
+++ b/pdns/dnsdistdist/m4/dnsdist_lua.m4
@@ -1,0 +1,39 @@
+AC_DEFUN([DNSDIST_LUA],[
+  AC_MSG_CHECKING([which version of Lua will be linked against])
+  AC_ARG_WITH([lua],
+    [AS_HELP_STRING([--with-lua], [Lua version to build against @<:@default=auto@:>@])],
+    [with_lua=$withval],
+    [with_lua=auto]
+  )
+  AC_MSG_RESULT([$with_lua])
+
+  AS_IF([test "x$with_lua" != "xno"],[
+    AS_IF([test "x$with_lua" = "xyes" -o "x$with_lua" = "xauto"],[
+      for LUAPC in lua5.3 lua-5.3 lua53 lua5.2 lua-5.2 lua52 lua5.1 lua-5.1 lua51 lua; do
+        PKG_CHECK_MODULES([LUA], $LUAPC >= 5.1, [
+          AC_DEFINE([HAVE_LUA], [1], [Define to 1 if you have lua])
+          with_lua=yes
+        ],[
+          LUAPC="" # otherwise pkg_check will fail
+        ])
+        if test "x$LUA_LIBS" != "x"; then break; fi
+      done
+    ],[
+       LUAPC="$with_lua"
+       PKG_CHECK_MODULES([LUA], $LUAPC >= 5.1, [
+         AC_DEFINE([HAVE_LUA], [1], [Define to 1 if you have Lua])
+         with_lua=yes
+       ])
+    ])
+    AC_MSG_CHECKING([for chosen LUA])
+    AS_IF([test "x$LUAPC" = "x"], [
+      AC_MSG_ERROR([no Lua found])
+      ],[
+      AC_MSG_RESULT([$LUAPC])
+    ])
+  ],[
+    AC_MSG_ERROR([Lua is not optional])
+  ])
+  AM_CONDITIONAL([LUA], [test "x$with_lua" = "xyes"])
+])
+

--- a/pdns/dnsdistdist/populate
+++ b/pdns/dnsdistdist/populate
@@ -16,7 +16,6 @@ ln -fs ../../../m4/pdns_check_libsodium.m4 m4/
 ln -fs ../../../m4/pdns_check_readline.m4 m4/
 ln -fs ../../../m4/ax_lib_readline.m4 m4/
 ln -fs ../../../m4/boost.m4 m4/
-ln -fs ../../../m4/pdns_with_lua.m4 m4/
 ln -fs ../../../m4/ax_cxx_compile_stdcxx_11.m4 m4/
 ln -fs ../../../m4/pdns_check_clock_gettime.m4 m4/
 

